design/cdk-bootstrap.md

```diff
@@ -0,0 +1,637 @@
+
+* An IAM role, called the **CloudFormation Execution Role**,
+  that will be used to perform the actual CFN stack deployment in the continuous delivery pipeline to this environment.
+  It is assumable *only* by the CloudFormation service principal
+  (this is for security reasons, as this role will have, necessarily, very wide permissions).
+  It will not have any inline policies,
+  but will instead have the Managed Policies attached that the user selected when prompted (see above).
+
+#### Physical resource names
+
+All of the above resources will be created with well-defined physical names -
+none of them will rely on automated CloudFormation naming.
+This is so that the other stages of CDK synthesis,
+like asset resolution, can rely on the concrete names
+(there is no reliable way to transfer this kind of information in an automated way across regions and/or accounts).
+This also allows for more fine-grained permissions -
+for instance, the continuous delivery pipeline needs to grant the **Deployment Action Role**
+permissions to read from the pipeline's bucket
+(to read the artifact that contains the CFN template to deploy);
+this way, it can add a well-defined name to the resource policy of the artifact bucket,
+instead of granting those permissions to all principals in the account.
+
+The naming scheme will include the following elements in order to minimize the chance of name collisions:
+
+* The region we're bootstrapping in.
+* The account ID we're bootstrapping in.
+* The type of the resource (file assets bucket, Docker assets repository, etc.).
+
+### `--trust`
+
+We will add a new command-line option called `--trust` to the `bootstrap` command.
+Its value is a list of AWS account IDs:
+
+```shell
+$ cdk bootstrap \
+    [--trust accountId[,otherAccountId]...] \
+    [--cloudformation-execution-policies policyName[,otherPolicyName]...] \
+    aws://account/region
+```
+
+If a user wants to add new trusted account(s) to an existing bootstrap stack,
+they have to specify all of the accounts they want to trust in the `--trust` option,
+not only the one being newly added -
+otherwise, the previously trusted account(s) will be removed.
+
+We will also add a another option,
+`--cloudformation-execution-policies`,
+that allows you to pass a list of managed policy ARNs on the command line to attach to the
+**CloudFormation Execution Role**.
+
+### Removing existing customization options
+
+The existing customization options: `--bootstrap-bucket-name` and `--bootstrap-kms-key-id` will be removed.
+We will need to know the names of the bootstrap bucket and KMS key and synthesis time.
+The only way to customize the bootstrap template will be to deploy your own,
+based on the default one the CDK provides,
+and then change the default options when creating instances of the `Stack`
+class to match the names used in the custom template.
+
+### CLI options in detail
+
+These options are inherited from the current CLI experience,
```

Comment:
I don't think it's confusing honestly - the "end state" is exactly this, "current state" + "what is going to be removed and what is going to be added".

If I only described the end state, it would be very unobvious that we're removing some CLI options, for example.